justfile

```diff
@@ -3,32 +3,46 @@
 @_default:
     just --list --unsorted
 
-# Run all tests
-test:
-    # These are the same tests that are run on CI. Eventually CI should just call into justfile
+# Run cargo check
+check:
     cargo check
+
+_add_tools:
     rustup component add clippy rustfmt
-    cargo fmt --all -- --check
-    cargo clippy --all-targets --all-features -- -D warnings
-    cargo test --all-targets --all-features
+
+# Run all tests
+test:
     cargo test --features http-async
     cargo test --features mmap-async-tokio
     cargo test --features tilejson
     cargo test --features s3-async-native
     cargo test --features s3-async-rustls
+    cargo test --features aws-s3-async
     cargo test
     RUSTDOCFLAGS="-D warnings" cargo doc --no-deps
 
+# Run all tests and checks
+test-all: check fmt clippy
+
 # Run cargo fmt and cargo clippy
 lint: fmt clippy
 
 # Run cargo fmt
-fmt:
+fmt: _add_tools
+    cargo fmt --all -- --check
+
+# Run cargo fmt using Rust nightly
+fmt-nightly:
     cargo +nightly fmt -- --config imports_granularity=Module,group_imports=StdExternalCrate
 
 # Run cargo clippy
-clippy:
-    cargo clippy --workspace --all-targets --all-features --bins --tests --lib --benches -- -D warnings
+clippy: _add_tools
+    cargo clippy --features http-async
```

Comment:
I would always include `--workspace --all-targets` params (not the other ones like `--bins` because that's redundant with `--all-targets`) -- this way these just recipes are easily copyable between projects, and are consistent